### PR TITLE
Limit map another block prompt

### LINKED ISSUE #802: User can be in Treecorder without having reservations.
If a user surveys the last block they have reserved and selects "Yes" to mapping more, they are taken back to the first survey page where they are told to select a block to map. No blocks are selectable since they have no reservations. They should probably go to the reservations page instead with a message about needing to reserve more blocks.

### PR DESCRIPTION
**Pull request by steventlamb**
fixes #802

Only show map another block prompt when there are remaining reservations.

### CODE REVIEW

File: src/nyc_trees/apps/survey/templates/survey/partials/map_another_popup.html [hunk 1]
```diff
@@ -2,15 +2,23 @@
...
-                <a id="map-another-yes" href="{% url 'survey' %}" class="btn btn-primary">
-                    Yes!
+                <a id="map-another-no"
+                   class="btn btn-default"
+                   data-class="map-another-possible"
+                   href="{% url 'progress_page' %}">I'm Done For Now</span>
+                <a id="map-another-no"
+                   class="btn btn-default"
+                   data-class="map-another-impossible"
+                   href="{% url 'progress_page' %}">OK</span>
```
Comment (steventlamb) [inline src/nyc_trees/apps/survey/templates/survey/partials/map_another_popup.html]: @dboyer @jfrankl  This needs content review. How does this look?

screenshot from 2015-03-10 18 59 26

File: src/nyc_trees/apps/survey/templates/survey/partials/map_another_popup.html [hunk 2]
```diff
@@ -2,15 +2,23 @@
...
-                <a id="map-another-no" href="{% url 'progress_page' %}" class="btn btn-default">
-                    I'm Done For Now
-                </a>
-                <a id="map-another-yes" href="{% url 'survey' %}" class="btn btn-primary">
-                    Yes!
+                <a id="map-another-no"
+                   class="btn btn-default"
+                   data-class="map-another-possible"
+                   href="{% url 'progress_page' %}">I'm Done For Now</span>
+                <a id="map-another-no"
```
Comment (maurizi) [inline src/nyc_trees/apps/survey/templates/survey/partials/map_another_popup.html]: This id is the same as the element above.  We shouldn't have duplicate ids

**Comment by maurizi** [comment]
I don't see anything in the view code that differentiates between surveys for events and surveys for reservations.

We shouldn't be checking remaining reservations when the survey was submitted for an event.

**Comment by steventlamb** [comment]
@maurizi ready for another look.

**Comment by maurizi** [comment]
@azavea-bot rebuild

**Comment by dboyer** [comment]
Let's add a note: You've mapped all your reserved blocks!

**Pull request merged**